addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4672,6 +4672,95 @@ registry.ReplaceMedia = SnippetOptionWidget.extend({
         this.$target.dblclick();
     },
 });
+ 
+/**
+ * Allows the image to redirect to custom URL on click.
+ */
+registry.LinkTool = SnippetOptionWidget.extend({
+
+    /**
+     * @override
+     */
+    start() {
+        this.linkEl = this.$target[0].parentElement.tagName === 'A' ?
+            this.$target[0].parentElement : null;
+        this.isNewWindow = this.linkEl ?
+            this.linkEl.getAttribute('target') === '_blank' : false;
+        return this._super(...arguments);
+    },
+    // OPTIONS TODO
+
+    /**
+     * Makes that when the link is clicked, the url will open on another page. This function
+     * is also called for the opposite operation (make the url open in the current page)
+     *
+     * @see this.selectClass for parameters
+     */
+    setNewWindow() {
+        this.isNewWindow = !this.isNewWindow;
+        if (this.isNewWindow) {
+            this.linkEl.setAttribute('target', '_blank');
+        } else {
+            this.linkEl.removeAttribute('target');
+        }
+    },
+    /**
+     * Records the target url of the link.
+     *
+     * @see this.selectClass for parameters
+     */
+    setUrl(previewMode, url) {
+        if (url && !url.startsWith('/') && !url.startsWith('#') && !url.startsWith('http')) {
+            url = 'http://' + url;
+        }
+        if (!url && !this.linkEl) {
+            return;
+        }
+        if (this.linkEl && !url) {
+            // Remove the linkEl
+            this.$target[0].parentNode.replaceWith(this.$target[0]);
+            this.linkEl = null;
+            this.isNewWindow = false;
+            this.updateUI(); // In order to hide the "Open in a new tab" field
```

Comment:
You'll never need to call `updateUI` by yourself, it is called automatically after each option is applied.

Reply:
I replaced it with `updateUIVisibility`... I don't know if I have the "right" to do that.
I do this so that the we-select that offers the anchors possibilities is visible before the input URL loses focus.
(and that the user sees it before moving on to edit something else or save)
I see 3 possibilities at the moment: 
1. We remove this `updateUIVisibility` and the user will have to make the input URL lose focus to see the we-select appear
2. I remove this `updateUIVisibility` and bypass the `computeWidgetVisibility` system by adding/removing the `d-none` class on the `we-select`.
3. We keep this `updateUIVisibility` that I run only under this condition finally:
```
if ((oldHref.length === 0 && url.length > 0) || (oldHref.length > 0 && url.length === 0)) {
            this.updateUIVisibility();
}
```

Reply:
Choice number 1. But you won't have to lose focus to see the select appear as the input is supposed to open an autocomplete list of URLs to choose and/or you could hit "enter" to confim your URL. But even without that, the page anchor selection is a plus, it's not an important feature, you don't need to see it first while you are typing the URL. If you did: then it should simply not be hidden ever.